### PR TITLE
JR-802-job-search-lang-change

### PR DESCRIPTION
**Pull request by wwebec**

### CODE REVIEW

File: src/main/webapp/app/job-search/state-management/effects/job-search.effects.ts [hunk 1]
```diff
@@ -119,6 +122,26 @@ export class JobSearchEffects {
...
 
+    @Effect()
+    languageChange$: Observable<Action> = this.actions$
+        .ofType(LANGUAGE_CHANGED)
+        .withLatestFrom(this.store.select(getSearchQuery))
+        .filter(([action, state]) => !!state.occupations)
+        .switchMap(([action, state]) => this.mapActionAndStateToUpdateOccupationTranslationAction(action, state));
+
+    private mapActionAndStateToUpdateOccupationTranslationAction(action: Action, state: any) {
+        const {occupations} = state;
```
Comment (birom) [inline src/main/webapp/app/job-search/state-management/effects/job-search.effects.ts]: Please fix the tslint warnings, or at least do not introduce new ones..
```
WARNING in ./src/main/webapp/app/job-search/state-management/effects/job-search.effects.ts
[133, 15]: A space is required after '{'
[133, 27]: A space is required before '}'

```

File: src/main/webapp/app/home/state-management/state/job-search-tool.state.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { TypeaheadMultiselectModel } from '../../../shared/input-components';
 import { ONLINE_SINCE_DEFAULT_VALUE } from '../../../shared/constants/job-search.constants';
 
 export interface JobSearchToolState {
+    occupations?: Array<TypeaheadMultiselectModel>;
```
Comment (birom) [inline src/main/webapp/app/home/state-management/state/job-search-tool.state.ts]: Earlier we had the baseQuery that is a general holder for free text values, occupations and classifications.
see: job-search-reducer.spec.ts.

I think we either should keep just the baseQuery or introduce a variable for free texts, occupations and classifications as well and do not use the baseQuery anymore. I find this "hybrid" solution with baseQuery and occupations in the same time confusing. We can have different translations only for classifications anyway.

Comment (birom) [inline src/main/webapp/app/home/state-management/state/job-search-tool.state.ts]: The selected values from on the landing page must be carried to the job search page if the user clicks on the "Show job offers button". These changes break this feature. Please have a closer look at the [job-search.effects.ts](https://github.com/alv-ch/jobroom2/blob/develop/src/main/webapp/app/job-search/state-management/effects/job-search.effects.ts#L62) and the [search-request-mapper.ts](https://github.com/alv-ch/jobroom2/blob/develop/src/main/webapp/app/job-search/state-management/util/search-request-mapper.ts#L20)

Comment (wwebec) [reply src/main/webapp/app/home/state-management/state/job-search-tool.state.ts]: Carrying of selected values has been corrected.

File: src/main/webapp/app/job-search/state-management/reducers/job-search.reducers.ts [hunk 3]
```diff
@@ -82,6 +83,11 @@ export function jobSearchReducer(state = initialState, action: Actions | core.La
                 )
             });
             break;
+        case UPDATE_OCCUPATION_TRANSLATION:
+            newState = Object.assign({}, state, {
+                    searchQuery: Object.assign({}, state.searchQuery, { baseQuery: action.payload })
+                });
```
Comment (Invizible) [inline src/main/webapp/app/job-search/state-management/reducers/job-search.reducers.ts]: formatting is broken here

File: src/main/webapp/app/home/state-management/effects/home.effects.ts [hunk 4]
```diff
@@ -110,4 +124,22 @@ export class HomeEffects {
...
+
+        return Observable.forkJoin(translations)
+            .map((translatedOccupations: Array<TypeaheadMultiselectModel>) =>
+                new JobSearchUpdateOccupationTranslationAction(translatedOccupations));
+    }
+
+    private translateIfOccupationHasTypeClassification(occupation: TypeaheadMultiselectModel, language: string): Observable<TypeaheadMultiselectModel> {
+        return occupation.type === OCCUPATION_CLASSIFICATION ?
+            this.occupationPresentationService.findOccupationLabelsByCode(occupation.code, language)
+                .map((label) => new TypeaheadMultiselectModel(occupation.type, occupation.code, label.default))
```
Comment (varusakov) [inline src/main/webapp/app/home/state-management/effects/home.effects.ts]: I propose to add error handling here. If one request fails, `Observable.forkJoin` will lost all values.

File: src/main/webapp/app/home/state-management/index.ts [hunk 5]
```diff
@@ -44,4 +48,5 @@ export type Actions =
     | SelectAgencyTabAction
     | GetActiveSystemNotificationsAction
     | GetActiveSystemNotificationsSuccessAction
-    | GetActiveSystemNotificationsFailedAction;
+    | GetActiveSystemNotificationsFailedAction
+    | JobSearchUpdateOccupationTranslationAction;
```
Comment (varusakov) [inline src/main/webapp/app/home/state-management/index.ts]: Add new line at the end of file

Comment (wwebec) [reply src/main/webapp/app/home/state-management/index.ts]: added

File: src/main/webapp/app/dashboard/pea-dashboard/pea-dashboard.component.ts [hunk 6]
```diff
@@ -1,127 +0,0 @@
-import {
```
Comment (varusakov) [inline src/main/webapp/app/dashboard/pea-dashboard/pea-dashboard.component.ts]: Why this file was deleted?

**Pull request merged**